modules/launcher/src/main/java/org/mule/module/launcher/ConfigurationManagamentAware.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.launcher;
+
+import static java.io.File.separator;
+import static org.mule.util.FileUtils.newFile;
+
+import org.mule.api.MuleRuntimeException;
+import org.mule.config.i18n.CoreMessages;
+import org.mule.config.i18n.Message;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Properties;
+
+public class ConfigurationManagamentAware
+{
+    private static final String CONFIGURATION_MANAGEMENT_FILE_NAME = "configuration-management.properties";
+
+    protected Properties manageConfigurationManagementPersistence(String workingDirectory, Properties configurationManagementProperties) throws IOException
```

Comment:
javadoc. Explain parameters. Is workingDirectory an specific folder per app? Where is located?